common/src/main/java/org/keycloak/common/enums/SslRequired.java

```diff
@@ -35,23 +35,29 @@ public boolean isRequired(ClientConnection connection) {
         return isRequired(connection.getRemoteAddr());
     }
 
-    public boolean isRequired(String address) {
+    public boolean isRequired(String host) {
         switch (this) {
             case ALL:
                 return true;
             case NONE:
                 return false;
             case EXTERNAL:
-                return !isLocal(address);
+                // NOTE: this is sometimes using hostnames here, which require DNS resolution
+                // It assumes that the resolution will be the same on the client side
+                // - this will go away once EXTERNAL is no longer supported
+                return !isLocal(host);
             default:
                 return true;
         }
     }
 
-    private boolean isLocal(String remoteAddress) {
+    private boolean isLocal(String host) {
+        if (host == null || host.isEmpty()) {
+            return false; // InetAddress.getByName returns localhost for these
+        }
         try {
-            InetAddress inetAddress = InetAddress.getByName(remoteAddress);
-            return inetAddress.isAnyLocalAddress() || inetAddress.isLoopbackAddress() || inetAddress.isSiteLocalAddress() || inetAddress.isLinkLocalAddress() || isUniqueLocal(inetAddress);
+            InetAddress inetAddress = InetAddress.getByName(host);
```

Comment:
While we're at it I would consider doing our own thing based on CIDR ranges, or at least use `getByAddress` to prevent any host lookups.

Reply:
See the response to the other comment where we're using this in other places with hostnames.